application/common/models/User.php

```diff
@@ -426,13 +435,42 @@ public function getDisplayName()
     }
 
     /**
-     * Is it time to nag the user to add an MFA?
-     *
-     * @return bool
+     * Based on current time and presence of MFA and Method options,
+     * determine which "nag" to present to the user.
      */
-    public function isTimeToNagForMfa()
+    public function getNagState()
```

Comment:
I find the logic of this method a little hard to follow. What is our order of priority for which nag (when applicable) should happen?

I wonder if there's a way to use an array of closures keyed on NAG_* constants, where we could just use the first one that returns true. Something like the following (though my example code's syntax might not be valid):
```php
public function getNagState()
{
    if ($this->nagState !== null) {
        return $this->nagState;
    }
    
    $possibleNags = [
        self::NAG_ADD_MFA => function isTimeToNagToAddMfa() { ... },
        self::NAG_REVIEW_MFA => function isTimeToNagToReviewMfa() { ... },
        self::NAG_ADD_METHOD => function isTimeToNagToAddMethod() { ... },
        self::NAG_REVIEW_METHOD => function isTimeToNagToReviewMethod() { ... },
        self::NAG_NONE => true,
    ];
    
    foreach ($possibleNags as $nagType => $isTime) {
        if ($isTime()) {
            $this->nagState = $nagType;
            return $this->nagState;
        }
    }
}
```

Reply:
Good idea. I'll see if it works.

Reply:
Yep. Worked perfectly.